zkvm/src/types.rs

```diff
@@ -59,6 +59,12 @@ pub enum Data {
 
     /// An input object (claimed UTXO).
     Output(Box<Output>),
+
+    /// A call proof.
+    CallProof(CallProof),
```

Comment:
Not sure that there's value to hold the unparsed `Data::CallProof` variant. It does not contain any secret data, and encoding/decoding it on prover's end won't make a performance difference. For simplicity, i'd remove it and use `Data::Opaque` which must be parsed into CallProof both by prover and verifier VMs.